packages/fuels-contract/src/contract.rs

```diff
@@ -550,14 +581,23 @@ where
         let attempts = max_attempts.unwrap_or(DEFAULT_TX_DEP_ESTIMATION_ATTEMPTS);
 
         for _ in 0..attempts {
-            let result = self.call_or_simulate(true).await;
+            let result = self.simulate_without_decode().await;
 
             match result {
                 Err(Error::RevertTransactionError(_, receipts))
                     if ContractCall::is_missing_output_variables(&receipts) =>
                 {
                     self = self.append_variable_outputs(1);
                 }
+
+                Err(Error::RevertTransactionError(_, receipts)) => {
```

Comment:
At a glance, given this branching structure, if an `Error::RevertTransactionError` occurs that is neither due to missing variables or a missing contract, the loop will continue. Instead, the error should be propagated.